Framework/API/test/AlgorithmTest.h

```diff
@@ -787,6 +806,86 @@ class AlgorithmTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(algCastConst, algCastNonConst);
   }
 
+  void testIndexingAlgorithm_declareIndexPropertyMethod() {
+    IndexingAlgorithm indexAlg;
+    TS_ASSERT_THROWS_NOTHING(
+        indexAlg.declareIndexProperty<MatrixWorkspace>("InputWorkspace"));
+  }
+
+  void testIndexingAlgorithm_setIndexPropertyMethods() {
+    IndexingAlgorithm indexAlg;
+    auto wksp =
+        WorkspaceFactory::Instance().create("WorkspaceTester", 10, 10, 9);
+    indexAlg.declareIndexProperty<MatrixWorkspace>("InputWorkspace1");
+    indexAlg.declareIndexProperty<MatrixWorkspace>("InputWorkspace2");
+    indexAlg.declareIndexProperty<MatrixWorkspace>("InputWorkspace3");
+    indexAlg.declareIndexProperty<MatrixWorkspace>("InputWorkspace4");
+
+    AnalysisDataService::Instance().add("wksp", wksp);
+    TS_ASSERT_THROWS_NOTHING(indexAlg.setIndexProperty<MatrixWorkspace>(
+        "InputWorkspace1", wksp, IndexType::WorkspaceIndex,
+        std::vector<int>{1, 2, 3, 4, 5}));
+    TS_ASSERT_THROWS_NOTHING(indexAlg.setIndexProperty<MatrixWorkspace>(
+        "InputWorkspace2", wksp, IndexType::WorkspaceIndex, "1:5"));
+    TS_ASSERT_THROWS_NOTHING(indexAlg.setIndexProperty<MatrixWorkspace>(
+        "InputWorkspace3", "wksp", IndexType::WorkspaceIndex,
+        std::vector<int>{1, 2, 3, 4, 5}));
+    TS_ASSERT_THROWS_NOTHING(indexAlg.setIndexProperty<MatrixWorkspace>(
+        "InputWorkspace4", "wksp", IndexType::WorkspaceIndex, "1:5"));
+    AnalysisDataService::Instance().remove("wksp");
+  }
+
+  void testIndexingAlgorithm_getIndexPropertyMethod() {
+    IndexingAlgorithm indexAlg;
+    indexAlg.declareIndexProperty<MatrixWorkspace>("InputWorkspace");
+
+    auto wksp =
+        WorkspaceFactory::Instance().create("WorkspaceTester", 10, 10, 9);
+    AnalysisDataService::Instance().add("wksp", wksp);
+    indexAlg.setIndexProperty<MatrixWorkspace>(
+        "InputWorkspace", "wksp", IndexType::WorkspaceIndex, "1:5");
+
+    MatrixWorkspace_sptr wsTest;
+    Indexing::SpectrumIndexSet indexSet(0);
+
+    TS_ASSERT_THROWS_NOTHING(
+        std::tie(wsTest, indexSet) =
+            indexAlg.getIndexProperty<MatrixWorkspace>("InputWorkspace"));
```

Comment:
Do I understand correctly that this would now also work with C++17 structured bindings, without extra casting?
```cpp
auto [ws, indexSet] = indexAlg.getIndexProperty<MatrixWorkspace>("InputWorkspace"));
```